metadata-ingestion/src/datahub/ingestion/source/schema_inference/avro.py

```diff
@@ -0,0 +1,86 @@
+from io import TextIOWrapper
+from typing import List, Type
+
+import ujson
+from avro.datafile import DataFileReader
+from avro.io import DatumReader
+
+from datahub.ingestion.source.schema_inference.base import SchemaInferenceBase
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    ArrayTypeClass,
+    BooleanTypeClass,
+    BytesTypeClass,
+    EnumTypeClass,
+    MapTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    RecordTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    StringTypeClass,
+    UnionTypeClass,
+)
+
+# see https://avro.apache.org/docs/current/spec.html
+avro_type_map = {
+    "null": NullTypeClass,
+    "boolean": BooleanTypeClass,
+    "int": NumberTypeClass,
+    "long": NumberTypeClass,
+    "float": NumberTypeClass,
+    "double": NumberTypeClass,
+    "bytes": BytesTypeClass,
+    "string": StringTypeClass,
+    "record": RecordTypeClass,
+    "enum": EnumTypeClass,
+    "array": ArrayTypeClass,
+    "map": MapTypeClass,
+    "fixed": StringTypeClass,
+}
+
+
+class AvroInferrer(SchemaInferenceBase):
+    @staticmethod
+    def infer_schema(file: TextIOWrapper) -> List[SchemaField]:
+
+        reader = DataFileReader(file, DatumReader())
+        schema = ujson.loads(reader.schema)
+
+        fields = []
+
+        for field in schema["fields"]:
```

Comment:
Why is https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/extractor/schema_util.py#L461 not used to parse this schema?

Reply:
Didn't know we had this! Updated